reports/management/commands/generate_reports.py

```diff
@@ -134,601 +69,24 @@ def add_arguments(self, parser):
-        # instead gets her subscription changed to loss.
-        changes = self.get_changes(
-            created_at__gte=start_date.isoformat(),
-            created_at__lte=end_date.isoformat(), action='change_loss')
-        for change in changes:
-            identity = self.get_identity(ids_client, change.mother_id)
-            if identity.get('details', {}).get('linked_to'):
-                row = self.get_related_row(
-                    rows, identity['details']['linked_to'],
-                    change.created_at.isoformat)
-            else:
-                row = collections.defaultdict(list)
-            row['loss_subscription'] = 'Yes'
-            row['timestamp'] = change.created_at.isoformat()
-            row['reason'] = 'miscarriage'
-            if identity['details'].get('receiver_role'):
-                row['receivers'].append(identity['details']['receiver_role'])
-
-            # Get the message set. This is a best guess effort.
-            message_set = self.guess_message_set(
-                sbm_client, change.mother_id, change.created_at)
-            if message_set:
-                row['message_sets'].append(message_set)
-
-            # Try to get the registered receiver, this is a best guess effort.
-            registered_receiver = self.guess_registered_receiver(
-                change.mother_id, optout['created_at'])
-            if registered_receiver:
-                row['registered_receiver'] = registered_receiver
-
-            rows.append(row)
-
-        rows.sort(key=lambda r: r['timestamp'])
-        for row in rows:
-            if all(r in row['registered_reciever'] for r in row['receivers']):
-                optout_receiver = "All messages"
-            else:
-                try:
-                    optout_receiver = "{} messages".format(row['receivers'][0])
-                except IndexError:
-                    optout_receiver = ''
-            sheet.add_row({
-                "Timestamp": row['timestamp'] or '',
-                "Registered Receiver": row['registered_receiver'] or '',
-                "Opt Out Reason": row['reason'] or '',
-                "Loss Subscription": row['loss_subscription'] or '',
-                "Opt Out Receiver": optout_receiver,
-                "Message Sets": ', '.join(row['message_sets']),
-                "Receivers": ', '.join(row['receivers']),
-                "Number of Receivers": len(row['receivers']),
-            })
-
-        # Add a warning to the sheet, because we cannot guarantee the best
-        # guess results
-        sheet.add_row({
-            "Timestamp": (
-                "NOTE: This data is not guaranteed to be correct, as "
-                "the current structure of the data does not allow us to link "
-                "the opt out to a subscription or registration, so this is a "
-                "best-effort guess."),
+        generate_report.apply_async(kwargs={
```

Comment:
For the management command, do we maybe want it to rather be sync (ie. call the task directly, instead of have it go asyncrously through celery)?